include/benchmark/benchmark_api.h

```diff
@@ -243,6 +246,41 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 // FIXME Add ClobberMemory() for non-gnu compilers
 #endif
 
+
+
+// This class is used for user-defined counters.
+class Counter {
+public:
+
+  typedef enum {
+    kDefaults   = 0,
+    // Mark the counter as a rate. It will be presented divided
+    // by the duration of the benchmark.
+    kIsRate     = 1,
+    // Mark the counter as a thread-average quantity. It will be
+    // presented divided by the number of threads.
+    kAvgThreads = 2,
+    kAvgThreadsRate = kIsRate|kAvgThreads
+  } Flags;
+
+public:
+
+  double value;
+  Flags  flags;
+
+public:
+
+  BENCHMARK_ALWAYS_INLINE Counter(double v = 0., Flags f = kDefaults) : value(v), flags(f) {}
```

Comment:
watch your line-length please. you can always use clang-format with the Google style.